dev-docs/RFCs/v1.0/react-map-gl-draw.md

```diff
@@ -0,0 +1,156 @@
+# RFC: react-map-gl-draw
+
+## Background
+
+react-map-gl currently does not support drawing functions. However, we have got a couple of [users](https://github.com/uber/react-map-gl/issues/725) interested in this capability. Also it is one of P0 features on Kepler.gl 2019 [roadmap](https://github.com/uber/kepler.gl/wiki/Kepler.gl-2019-Roadmap#allow-drawing-on-map-to-create-paths-and-polygons--). 
+
+Although [Mapbox/mapbox-gl-draw](https://github.com/mapbox/mapbox-gl-draw) provides quite nice drawing and editing features, because of its manipulating internal states, it cannot work well with React / Redux  framework and therefore cannot be integrated with `react-map-gl`.
+[Nebula.gl](http://neb.gl) provides geo editing functionality, however it heavily depends on `deck.gl`.
+
+## Proposal
+
+`react-map-gl-draw` is a react based geo editing library, providing functions like the following.
+
+### Options 
+- `MapContextProvider` (React.Component, required) - A [Context.Provider](https://reactjs.org/docs/context.html#contextprovider) component. If supplied, will be rendered as the ancestor to all children. The passed through context contains the following values:
+  - `viewport` ([Viewport](/docs/api-reference/viewport.md)) - the current viewport
+  - `eventManager` ([EventManager](https://github.com/uber-web/mjolnir.js/blob/master/docs/api-reference/event-manager.md))
+
+- `mode` (String, Optional) - `react-map-gl` is stateless, user has complete control of the `mode`.
+  - `EditorModes.READ_ONLY` - Not interactive. This is the default mode.
+  - `EditorModes.SELECT_FEATURE` - Lets you select, delete, and drag features.
+  - `EditorModes.EDIT_VERTEX` - Lets you select, delete, and drag vertices; and drag features.
+  - `EditorModes.DRAW_PATH` - Lets you draw a LineString feature.
+  - `EditorModes.DRAW_POLYGON` - Lets you draw a Polygon feature.
+  - `EditorModes.DRAW_POINT` - Lets you draw a Point feature.
+  - `EditorModes.DRAW_RECTANGLE` - Lets you draw a Rectangle feature.
+
+- `getFeatureStyle` (Function, Optional) : Object - A function to style features, function parameters are 
```

Comment:
Have we decided what style properties we will support here?

Reply:
`react-map-gl-draw` is using `svg` `circle`, `rect` and `path` to draw features, so any styles which could be applied to these `svg` elements will be supported.

Reply:
Ohhh - I had it in my head that this was drawing to the gl context. Does this then create an `<svg>` element and overlay it on top of the canvas where Deck and/or Mapbox are drawing?

Reply:
Correct. All the drawing will be rendered in a svg overlay on top of react-map-gl base map